File: SRC.NET/Configuration.vb
Option Strict Off
Option Explicit On
Friend Class frmConfiguration
	Inherits System.Windows.Forms.Form
	
	' Copyright (C) 1997-2012 Kei Sakamoto / Inui Tetsuyuki
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	'Invalid_string_refer_to_original_code
	
	'繝槭ャ繝励さ繝槭Φ繝峨瑚ｨｭ螳壼､画峩縲咲畑繝繧､繧｢繝ｭ繧ｰ
	
	
	'MP3Volume繧定ｨ倬鹸
	Private SavedMP3Volume As Short
	
	'Invalid_string_refer_to_original_code
	'UPGRADE_WARNING: イベント chkBattleAnimation.CheckStateChanged は、フォームが初期化されたときに発生します。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"' をクリックしてください。
	Private Sub chkBattleAnimation_CheckStateChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles chkBattleAnimation.CheckStateChanged
		'Invalid_string_refer_to_original_code
		If chkBattleAnimation.CheckState = 1 Then
			chkExtendedAnimation.Enabled = True
			chkWeaponAnimation.Enabled = True
		Else
			chkExtendedAnimation.Enabled = False
			chkWeaponAnimation.Enabled = False
		End If
	End Sub
	
	'Invalid_string_refer_to_original_code
	Private Sub cmdCancel_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdCancel.Click
		Dim IsMP3Supported As Object
		'繝繧､繧｢繝ｭ繧ｰ繧帝哩縺倥ｋ
		Hide()
		
		'Invalid_string_refer_to_original_code
		MP3Volume = SavedMP3Volume
		'UPGRADE_WARNING: オブジェクト IsMP3Supported の既定プロパティを解決できませんでした。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"' をクリックしてください。
		If IsMP3Supported Then
			Call vbmp3_setVolume(MP3Volume, MP3Volume)
		End If
	End Sub
	
	'Invalid_string_refer_to_original_code
	Private Sub cmdOK_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdOK.Click
		'Invalid_string_refer_to_original_code
		
		'Invalid_string_refer_to_original_code
		Select Case cboMessageSpeed.Text
			Case "Invalid_string_refer_to_original_code"
				MessageWait = 0
			Case "Invalid_string_refer_to_original_code"
				MessageWait = 200
			Case "Invalid_string_refer_to_original_code"
				MessageWait = 400
			Case "Invalid_string_refer_to_original_code"
				MessageWait = 700
			Case "Invalid_string_refer_to_original_code"
				MessageWait = 1000
			Case "謇句虚騾√ｊ"
				MessageWait = 10000000
		End Select
		WriteIni("Option", "MessageWait", VB6.Format(MessageWait))
		
		'謌ｦ髣倥い繝九Γ陦ｨ遉ｺ
		If chkBattleAnimation.CheckState = 1 Then
			BattleAnimation = True
			WriteIni("Option", "BattleAnimation", "On")
		Else
			BattleAnimation = False
			WriteIni("Option", "BattleAnimation", "Off")
		End If
		
		'諡｡螟ｧ謌ｦ髣倥い繝九Γ陦ｨ遉ｺ
		If chkExtendedAnimation.CheckState = 1 Then
			ExtendedAnimation = True
			WriteIni("Option", "ExtendedAnimation", "On")
		Else
			ExtendedAnimation = False
			WriteIni("Option", "Extendednimation", "Off")
		End If
		
		'豁ｦ蝎ｨ貅門ｙ繧｢繝九Γ陦ｨ遉ｺ
		If chkWeaponAnimation.CheckState = 1 Then
			WeaponAnimation = True
			WriteIni("Option", "WeaponAnimation", "On")
		Else
			WeaponAnimation = False
			WriteIni("Option", "WeaponAnimation", "Off")
		End If
		
		'遘ｻ蜍輔い繝九Γ陦ｨ遉ｺ
		If chkMoveAnimation.CheckState = 1 Then
			MoveAnimation = True
			WriteIni("Option", "MoveAnimation", "On")
		Else
			MoveAnimation = False
			WriteIni("Option", "MoveAnimation", "Off")
		End If
		
		'繧ｹ繝壹す繝｣繝ｫ繝代Ρ繝ｼ繧｢繝九Γ陦ｨ遉ｺ
		If chkSpecialPowerAnimation.CheckState = 1 Then
			SpecialPowerAnimation = True
			WriteIni("Option", "SpecialPowerAnimation", "On")
		Else
			SpecialPowerAnimation = False
			WriteIni("Option", "SpecialPowerAnimation", "Off")
		End If
		
		'Invalid_string_refer_to_original_code
		If chkAutoMoveCursor.CheckState Then
			AutoMoveCursor = True
			WriteIni("Option", "AutoMoveCursor", "On")
		Else
			AutoMoveCursor = False
			WriteIni("Option", "AutoMoveCursor", "Off")
		End If
		
		'繝槭せ逶ｮ縺ｮ陦ｨ遉ｺ
		If chkShowSquareLine.CheckState Then
			ShowSquareLine = True
			WriteIni("Option", "Square", "On")
		Else
			ShowSquareLine = False
			WriteIni("Option", "Square", "Off")
		End If
		
		'蜻ｳ譁ｹ繝輔ぉ繧､繧ｺ髢句ｧ区凾縺ｮ繧ｿ繝ｼ繝ｳ陦ｨ遉ｺ
		If chkShowTurn.CheckState Then
			WriteIni("Option", "Turn", "On")
		Else
			WriteIni("Option", "Turn", "Off")
		End If
		
		'Invalid_string_refer_to_original_code
		If chkKeepEnemyBGM.CheckState Then
			KeepEnemyBGM = True
			WriteIni("Option", "KeepEnemyBGM", "On")
		Else
			KeepEnemyBGM = False
			WriteIni("Option", "KeepEnemyBGM", "Off")
		End If
		
		'MIDI貍泌･上↓DirectMusic繧剃ｽｿ逕ｨ縺吶ｋ
		If chkUseDirectMusic.CheckState Then
			WriteIni("Option", "UseDirectMusic", "On")
		Else
			WriteIni("Option", "UseDirectMusic", "Off")
		End If
		
		'Invalid_string_refer_to_original_code
		MidiResetType = cboMidiReset.Text
		WriteIni("Option", "MidiReset", (cboMidiReset.Text))
		
		'Invalid_string_refer_to_original_code
		WriteIni("Option", "MP3Volume", VB6.Format(MP3Volume))
		
		'繝繧､繧｢繝ｭ繧ｰ繧帝哩縺倥ｋ
		Hide()
	End Sub
	
	Private Sub frmConfiguration_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		'Invalid_string_refer_to_original_code
		
		'Invalid_string_refer_to_original_code
		cboMessageSpeed.Items.Add("謇句虚騾√ｊ")
		cboMessageSpeed.Items.Add("Invalid_string_refer_to_original_code")
		cboMessageSpeed.Items.Add("Invalid_string_refer_to_original_code")
		cboMessageSpeed.Items.Add("Invalid_string_refer_to_original_code")
		cboMessageSpeed.Items.Add("Invalid_string_refer_to_original_code")
		cboMessageSpeed.Items.Add("Invalid_string_refer_to_original_code")
		Select Case MessageWait
			Case 0
				cboMessageSpeed.Text = "Invalid_string_refer_to_original_code"
			Case 200
				cboMessageSpeed.Text = "Invalid_string_refer_to_original_code"
			Case 400
				cboMessageSpeed.Text = "Invalid_string_refer_to_original_code"
			Case 700
				cboMessageSpeed.Text = "Invalid_string_refer_to_original_code"
			Case 1000
				cboMessageSpeed.Text = "Invalid_string_refer_to_original_code"
			Case 10000000
				cboMessageSpeed.Text = "謇句虚騾√ｊ"
		End Select
		
		'謌ｦ髣倥い繝九Γ陦ｨ遉ｺ
		If BattleAnimation Then
			chkBattleAnimation.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkBattleAnimation.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		If Not FileExists(AppPath & "Lib\豎守畑謌ｦ髣倥い繝九Γ\include.eve") Then
			chkBattleAnimation.CheckState = System.Windows.Forms.CheckState.Indeterminate '辟｡蜉ｹ
		End If
		
		'諡｡蠑ｵ謌ｦ髣倥い繝九Γ陦ｨ遉ｺ
		If ExtendedAnimation Then
			chkExtendedAnimation.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkExtendedAnimation.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		If chkBattleAnimation.CheckState = 1 Then
			chkExtendedAnimation.Enabled = True
		Else
			chkExtendedAnimation.Enabled = False
		End If
		
		'豁ｦ蝎ｨ貅門ｙ繧｢繝九Γ陦ｨ遉ｺ
		If WeaponAnimation Then
			chkWeaponAnimation.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkWeaponAnimation.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		If chkBattleAnimation.CheckState = 1 Then
			chkWeaponAnimation.Enabled = True
		Else
			chkWeaponAnimation.Enabled = False
		End If
		
		'遘ｻ蜍輔い繝九Γ陦ｨ遉ｺ
		If MoveAnimation Then
			chkMoveAnimation.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkMoveAnimation.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'繧ｹ繝壹す繝｣繝ｫ繝代Ρ繝ｼ繧｢繝九Γ陦ｨ遉ｺ
		If SpecialPowerAnimation Then
			chkSpecialPowerAnimation.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkSpecialPowerAnimation.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'Invalid_string_refer_to_original_code
		If AutoMoveCursor Then
			chkAutoMoveCursor.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkAutoMoveCursor.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'繝槭せ逶ｮ縺ｮ陦ｨ遉ｺ
		If ShowSquareLine Then
			chkShowSquareLine.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkShowSquareLine.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'蜻ｳ譁ｹ繝輔ぉ繧､繧ｺ髢句ｧ区凾縺ｮ繧ｿ繝ｼ繝ｳ陦ｨ遉ｺ
		If LCase(ReadIni("Option", "Turn")) = "on" Then
			chkShowTurn.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkShowTurn.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'Invalid_string_refer_to_original_code
		If KeepEnemyBGM Then
			chkKeepEnemyBGM.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkKeepEnemyBGM.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'MIDI貍泌･上↓DirectMusic繧剃ｽｿ逕ｨ縺吶ｋ
		If LCase(ReadIni("Option", "UseDirectMusic")) = "on" Then
			chkUseDirectMusic.CheckState = System.Windows.Forms.CheckState.Checked
		Else
			chkUseDirectMusic.CheckState = System.Windows.Forms.CheckState.Unchecked
		End If
		
		'Invalid_string_refer_to_original_code
		cboMidiReset.Items.Add("None")
		cboMidiReset.Items.Add("GM")
		cboMidiReset.Items.Add("GS")
		cboMidiReset.Items.Add("XG")
		cboMidiReset.Text = MidiResetType
		
		'Invalid_string_refer_to_original_code
		SavedMP3Volume = MP3Volume
		txtMP3Volume.Text = VB6.Format(MP3Volume)
	End Sub
	
	'MP3髻ｳ驥丞､画峩
	'UPGRADE_NOTE: hscMP3Volume.Change はイベントからプロシージャに変更されました。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="4E2DC008-5EDA-4547-8317-C9316952674F"' をクリックしてください。
	'UPGRADE_WARNING: HScrollBar イベント hscMP3Volume.Change には新しい動作が含まれます。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6BA9B8D2-2A32-4B6E-8D36-44949974A5B4"' をクリックしてください。
	Private Sub hscMP3Volume_Change(ByVal newScrollValue As Integer)
		Dim IsMP3Supported As Object
		MP3Volume = newScrollValue
		txtMP3Volume.Text = VB6.Format(MP3Volume)
		'UPGRADE_WARNING: オブジェクト IsMP3Supported の既定プロパティを解決できませんでした。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"' をクリックしてください。
		If IsMP3Supported Then
			Call vbmp3_setVolume(MP3Volume, MP3Volume)
		End If
	End Sub
	
	'UPGRADE_NOTE: hscMP3Volume.Scroll はイベントからプロシージャに変更されました。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="4E2DC008-5EDA-4547-8317-C9316952674F"' をクリックしてください。
	Private Sub hscMP3Volume_Scroll_Renamed(ByVal newScrollValue As Integer)
		Dim IsMP3Supported As Object
		MP3Volume = newScrollValue
		txtMP3Volume.Text = VB6.Format(MP3Volume)
		'UPGRADE_WARNING: オブジェクト IsMP3Supported の既定プロパティを解決できませんでした。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"' をクリックしてください。
		If IsMP3Supported Then
			Call vbmp3_setVolume(MP3Volume, MP3Volume)
		End If
	End Sub
	
	'UPGRADE_WARNING: イベント txtMP3Volume.TextChanged は、フォームが初期化されたときに発生します。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"' をクリックしてください。
	Private Sub txtMP3Volume_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtMP3Volume.TextChanged
		Dim IsMP3Supported As Object
		MP3Volume = StrToLng((txtMP3Volume.Text))
		
		If MP3Volume < 0 Then
			MP3Volume = 0
			txtMP3Volume.Text = "0"
		ElseIf MP3Volume > 100 Then 
			MP3Volume = 100
			txtMP3Volume.Text = "100"
		End If
		
		hscMP3Volume.Value = MP3Volume
		
		'UPGRADE_WARNING: オブジェクト IsMP3Supported の既定プロパティを解決できませんでした。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"' をクリックしてください。
		If IsMP3Supported Then
			Call vbmp3_setVolume(MP3Volume, MP3Volume)
		End If
	End Sub
	Private Sub hscMP3Volume_Scroll(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.ScrollEventArgs) Handles hscMP3Volume.Scroll
		Select Case eventArgs.type
			Case System.Windows.Forms.ScrollEventType.ThumbTrack
				hscMP3Volume_Scroll_Renamed(eventArgs.newValue)
			Case System.Windows.Forms.ScrollEventType.EndScroll
				hscMP3Volume_Change(eventArgs.newValue)
		End Select
	End Sub
End Class

File: SRC.Sharp/SRCSharpForm/Forms/Configuration.cs
using System;
using System.Windows.Forms;
using SRCCore;
using SRCCore.Config;
using SRCCore.Lib;
using SRCCore.VB;

namespace SRCSharpForm
{
    internal partial class frmConfiguration : Form
    {

        // Copyright (C) 1997-2012 Kei Sakamoto / Inui Tetsuyuki
        // 本プログラムはフリーソフトであり、無保証です。
        // 本プログラムはGNU General Public License(Ver.3またはそれ以降)が定める条件の下で
        // 再頒布または改変することができます。

        // マップコマンド「設定変更」用ダイアログ


        // MP3Volumeを記録
        private float SavedMP3Volume;

        public SRCCore.SRC SRC;
        private Sound Sound => SRC.Sound;
        // XXX 型
        private LocalFileConfig SystemConfig => SRC.SystemConfig as LocalFileConfig;

        // 戦闘アニメOn・Off切り替え
        // UPGRADE_WARNING: イベント chkBattleAnimation.CheckStateChanged は、フォームが初期化されたときに発生します。 詳細については、'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"' をクリックしてください。
        private void chkBattleAnimation_CheckStateChanged(object eventSender, EventArgs eventArgs)
        {
            //// 戦闘アニメを表示しない場合は拡張戦闘アニメ、武器アニメ選択の項目を選択不能にする
            //if ((int)chkBattleAnimation.CheckState == 1)
            //{
            //    chkExtendedAnimation.Enabled = true;
            //    chkWeaponAnimation.Enabled = true;
            //}
            //else
            //{
            //    chkExtendedAnimation.Enabled = false;
            //    chkWeaponAnimation.Enabled = false;
            //}
        }

        // キャンセルボタンが押された
        private void cmdCancel_Click(object eventSender, EventArgs eventArgs)
        {
            // MP3音量のみその場で変更しているので元に戻す必要がある
            Sound.Player.SoundVolume = SavedMP3Volume;
            Close();
        }

        // OKボタンが押された
        private void cmdOK_Click(object eventSender, EventArgs eventArgs)
        {
            //// 各種設定を変更

            //// メッセージスピード
            //switch (cboMessageSpeed.Text ?? "")
            //{
            //    case "神の領域":
            //        {
            //            GUI.MessageWait = 0;
            //            break;
            //        }

            //    case "超高速":
            //        {
            //            GUI.MessageWait = 200;
            //            break;
            //        }

            //    case "高速":
            //        {
            //            GUI.MessageWait = 400;
            //            break;
            //        }

            //    case "普通":
            //        {
            //            GUI.MessageWait = 700;
            //            break;
            //        }

            //    case "低速":
            //        {
            //            GUI.MessageWait = 1000;
            //            break;
            //        }

            //    case "手動送り":
            //        {
            //            GUI.MessageWait = 10000000;
            //            break;
            //        }
            //}

            //SystemConfig.SetItem("Option", "MessageWait", SrcFormatter.Format(GUI.MessageWait));

            // 戦闘アニメ表示
            if ((int)chkBattleAnimation.CheckState == 1)
            {
                SystemConfig.BattleAnimation = true;
            }
            else
            {
                SystemConfig.BattleAnimation = false;
            }

            // 拡大戦闘アニメ表示
            if ((int)chkExtendedAnimation.CheckState == 1)
            {
                SystemConfig.ExtendedAnimation = true;
            }
            else
            {
                SystemConfig.ExtendedAnimation = false;
            }

            // 武器準備アニメ表示
            if ((int)chkWeaponAnimation.CheckState == 1)
            {
                SystemConfig.WeaponAnimation = true;
            }
            else
            {
                SystemConfig.WeaponAnimation = false;
            }

            // 移動アニメ表示
            if ((int)chkMoveAnimation.CheckState == 1)
            {
                SystemConfig.MoveAnimation = true;
            }
            else
            {
                SystemConfig.MoveAnimation = false;
            }

            // スペシャルパワーアニメ表示
            if ((int)chkSpecialPowerAnimation.CheckState == 1)
            {
                SystemConfig.SpecialPowerAnimation = true;
            }
            else
            {
                SystemConfig.SpecialPowerAnimation = false;
            }

            // マウスカーソルの自動移動
            if (chkAutoMoveCursor.Checked)
            {
                SystemConfig.AutoMoveCursor = true;
            }
            else
            {
                SystemConfig.AutoMoveCursor = false;
            }

            // マス目の表示
            if (chkShowSquareLine.Checked)
            {
                SystemConfig.ShowSquareLine = true;
            }
            else
            {
                SystemConfig.ShowSquareLine = false;
            }

            // 味方フェイズ開始時のターン表示
            if (chkShowTurn.Checked)
            {
                SystemConfig.SetItem("Option", "Turn", "On");
            }
            else
            {
                SystemConfig.SetItem("Option", "Turn", "Off");
            }

            // 敵フェイズ中にＢＧＭを変更しない
            if (chkKeepEnemyBGM.Checked)
            {
                SystemConfig.KeepEnemyBGM = true;
            }
            else
            {
                SystemConfig.KeepEnemyBGM = false;
            }

            //// MIDI音源リセットの種類
            //SystemConfig.MidiResetType = cboMidiReset.Text;
            //SystemConfig.SetItem("Option", "MidiReset", cboMidiReset.Text);
            //cboMidiReset.Text = argini_data21;

            // MP3再生音量
            SystemConfig.SoundVolume = (int)(Sound.Player.SoundVolume * 100);
            SystemConfig.Save();

            // ダイアログを閉じる
            Close();
        }

        private void frmConfiguration_Load(object eventSender, EventArgs eventArgs)
        {
            // ダイアログを初期化

            //// メッセージスピード
            //cboMessageSpeed.Items.Add("手動送り");
            //cboMessageSpeed.Items.Add("低速");
            //cboMessageSpeed.Items.Add("普通");
            //cboMessageSpeed.Items.Add("高速");
            //cboMessageSpeed.Items.Add("超高速");
            //cboMessageSpeed.Items.Add("神の領域");
            //switch (GUI.MessageWait)
            //{
            //    case 0:
            //        {
            //            cboMessageSpeed.Text = "神の領域";
            //            break;
            //        }

            //    case 200:
            //        {
            //            cboMessageSpeed.Text = "超高速";
            //            break;
            //        }

            //    case 400:
            //        {
            //            cboMessageSpeed.Text = "高速";
            //            break;
            //        }

            //    case 700:
            //        {
            //            cboMessageSpeed.Text = "普通";
            //            break;
            //        }

            //    case 1000:
            //        {
            //            cboMessageSpeed.Text = "低速";
            //            break;
            //        }

            //    case 10000000:
            //        {
            //            cboMessageSpeed.Text = "手動送り";
            //            break;
            //        }
            //}

            // 戦闘アニメ表示
            if (SRC.BattleAnimation)
            {
                chkBattleAnimation.CheckState = CheckState.Checked;
            }
            else
            {
                chkBattleAnimation.CheckState = CheckState.Unchecked;
            }

            //bool localFileExists() { string argfname = SRC.AppPath + @"Lib\汎用戦闘アニメ\include.eve"; 
            //    var ret = GeneralLib.FileExists(argfname); 
            //    return ret; }

            //if (!localFileExists())
            //{
            //    chkBattleAnimation.CheckState = CheckState.Indeterminate; // 無効
            //}

            // 拡張戦闘アニメ表示
            if (SRC.ExtendedAnimation)
            {
                chkExtendedAnimation.CheckState = CheckState.Checked;
            }
            else
            {
                chkExtendedAnimation.CheckState = CheckState.Unchecked;
            }

            if ((int)chkBattleAnimation.CheckState == 1)
            {
                chkExtendedAnimation.Enabled = true;
            }
            else
            {
                chkExtendedAnimation.Enabled = false;
            }

            // 武器準備アニメ表示
            if (SRC.WeaponAnimation)
            {
                chkWeaponAnimation.CheckState = CheckState.Checked;
            }
            else
            {
                chkWeaponAnimation.CheckState = CheckState.Unchecked;
            }

            if ((int)chkBattleAnimation.CheckState == 1)
            {
                chkWeaponAnimation.Enabled = true;
            }
            else
            {
                chkWeaponAnimation.Enabled = false;
            }

            // 移動アニメ表示
            if (SRC.MoveAnimation)
            {
                chkMoveAnimation.CheckState = CheckState.Checked;
            }
            else
            {
                chkMoveAnimation.CheckState = CheckState.Unchecked;
            }

            // スペシャルパワーアニメ表示
            if (SRC.SpecialPowerAnimation)
            {
                chkSpecialPowerAnimation.CheckState = CheckState.Checked;
            }
            else
            {
                chkSpecialPowerAnimation.CheckState = CheckState.Unchecked;
            }

            // マウスカーソルの自動移動
            if (SRC.AutoMoveCursor)
            {
                chkAutoMoveCursor.CheckState = CheckState.Checked;
            }
            else
            {
                chkAutoMoveCursor.CheckState = CheckState.Unchecked;
            }

            // マス目の表示
            if (SRC.ShowSquareLine)
            {
                chkShowSquareLine.CheckState = CheckState.Checked;
            }
            else
            {
                chkShowSquareLine.CheckState = CheckState.Unchecked;
            }

            //// 味方フェイズ開始時のターン表示
            //if (Strings.LCase(GeneralLib.ReadIni("Option", "Turn")) == "on")
            //{
            //    chkShowTurn.CheckState = CheckState.Checked;
            //}
            //else
            //{
            //    chkShowTurn.CheckState = CheckState.Unchecked;
            //}

            //// 敵フェイズ中にＢＧＭを変更しない
            //if (SRC.KeepEnemyBGM)
            //{
            //    chkKeepEnemyBGM.CheckState = CheckState.Checked;
            //}
            //else
            //{
            //    chkKeepEnemyBGM.CheckState = CheckState.Unchecked;
            //}

            //// MIDI演奏にDirectMusicを使用する
            //if (Strings.LCase(GeneralLib.ReadIni("Option", "UseDirectMusic")) == "on")
            //{
            //    chkUseDirectMusic.CheckState = CheckState.Checked;
            //}
            //else
            //{
            //    chkUseDirectMusic.CheckState = CheckState.Unchecked;
            //}

            //// MIDI音源リセットの種類
            //cboMidiReset.Items.Add("None");
            //cboMidiReset.Items.Add("GM");
            //cboMidiReset.Items.Add("GS");
            //cboMidiReset.Items.Add("XG");
            //cboMidiReset.Text = SRC.MidiResetType;

            // MP3音量
            SavedMP3Volume = Sound.Player.SoundVolume;
            txtMP3Volume.Text = SrcFormatter.Format((int)(Sound.Player.SoundVolume * 100));
        }

        // MP3音量変更
        private void hscMP3Volume_Change(int newScrollValue)
        {
            Sound.Player.SoundVolume = newScrollValue / 100f;
            txtMP3Volume.Text = SrcFormatter.Format(newScrollValue);
        }

        private void hscMP3Volume_Scroll(int newScrollValue)
        {
            Sound.Player.SoundVolume = newScrollValue / 100f;
            txtMP3Volume.Text = SrcFormatter.Format(newScrollValue);
        }

        private void txtMP3Volume_TextChanged(object eventSender, EventArgs eventArgs)
        {
            var volume = GeneralLib.StrToLng(txtMP3Volume.Text);
            if (volume < 0)
            {
                volume = 0;
                txtMP3Volume.Text = "0";
            }
            else if (volume > 100)
            {
                volume = 100;
                txtMP3Volume.Text = "100";
            }

            hscMP3Volume.Value = volume;
        }

        private void hscMP3Volume_Scroll(object eventSender, ScrollEventArgs eventArgs)
        {
            switch (eventArgs.Type)
            {
                case ScrollEventType.ThumbTrack:
                    {
                        hscMP3Volume_Scroll(eventArgs.NewValue);
                        break;
                    }

                case ScrollEventType.EndScroll:
                    {
                        hscMP3Volume_Change(eventArgs.NewValue);
                        break;
                    }
            }
        }
    }
}
